### PR TITLE
New resource and data source `outlook_category`

### PR DESCRIPTION
**Pull request by ArcturusZhang**

### CODE REVIEW

**Comment by ArcturusZhang** [comment]
Please hold on merging. I did not get time to run the acc tests yet

**Comment by ArcturusZhang** [comment]
@magodo Test has passed
```
=== RUN   TestAccOutlookCategory_basic
--- PASS: TestAccOutlookCategory_basic (5.63s)
=== RUN   TestAccOutlookCategory_update
--- PASS: TestAccOutlookCategory_update (8.53s)
PASS
ok      github.com/magodo/terraform-provider-outlook/outlook/services   14.203s
```

File: outlook/services/category_resource.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
```
Comment (magodo) [inline outlook/services/category_resource.go:37]: Shall we make it `Forcenew`?

File: website/docs/r/category.html.markdown [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+* `read` - (Defaults to 5 minutes) Used when retrieving the Category.
+* `update` - (Defaults to 30 minutes) Used when updating the Category.
+* `delete` - (Defaults to 30 minutes) Used when deleting the Category.
+
+## Import
+
+Categories can be imported using the `resource id`, e.g.
+
+```shell
+terraform import outlook_message_rule.example <id>
```
Comment (magodo) [inline website/docs/r/category.html.markdown]:
```suggestion
terraform import outlook_category.example <id>
```

**Review by magodo — CHANGES_REQUESTED** [review]
Thanks for this PR!
I have left some additional mostly minor comments that once addressed this should be good to merge 👍

File: website/docs/r/category.html.markdown (hunk 2 above)
Comment (ArcturusZhang) [reply website/docs/r/category.html.markdown]: Oops
Done

File: outlook/services/category_resource.go (hunk 1 above)
Comment (ArcturusZhang) [reply outlook/services/category_resource.go:37]: Done

**Comment by ArcturusZhang** [comment]
Hi @magodo I have resolved the comments and the conflicts, please have a look.

**Pull request merged**